library/manageiq.py

```diff
@@ -64,14 +61,110 @@
 '''
 
 
-def update_required(client, module, providers_url, provider_id, hostname, port):
-    try:
-        result = client.get(providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
+class ManageIQ(object):
+    """ ManageIQ object to execute various operations in manageiq
+
+    url      - manageiq environment url
+    user     - the username in manageiq
+    password - the user password in manageiq
+    """
+    openshift_provider_type = 'ManageIQ::Providers::OpenshiftEnterprise::ContainerManager'
+
+    def __init__(self, module, url, user, password):
+        self.module        = module
+        self.api_url       = url + '/api'
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.api_url, (self.user, self.password))
+        self.changed       = False
+        self.msg           = ''
+        self.providers_url = self.api_url + '/providers'
+
+    def update_required(self, provider_id, hostname, port):
+        """ Checks whether an update is required for the provider
+
+        Returns:
+            False if the hostname and port passed equals the provider's,
+            True otherwise
+        """
+        try:
+            result = self.client.get(self.providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
+        except Exception as e:
+            self.msg = "Failed to get provider data. Error: %s" % e
+            self.module.fail_json(msg=self.msg)
```

Comment:
Agreed.
`fail_json()` does `sys.exit(1)`; setting an ivar gives false impression it matters ("is it used later?  does fail_json() return?"), so is slightly confusing to read.